.cbmc-batch/jobs/aws_array_list_comparator_string/aws_array_list_comparator_string_harness.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <aws/common/string.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+#include <stddef.h>
+
+const size_t MAX_STRING_LEN = 16;
+
+/**
+ * Coverage: 1.00 (56 lines out of 56 statically-reachable lines in 10 functions reached)
+ * Runtime: real	0m4.682s
+ */
+void aws_array_list_comparator_string_harness() {
+    struct aws_string *str_a = make_arbitrary_aws_string_nondet_len_with_max(can_fail_allocator(), MAX_STRING_LEN);
+    struct aws_string *str_b =
+        nondet_bool() ? str_a : make_arbitrary_aws_string_nondet_len_with_max(can_fail_allocator(), MAX_STRING_LEN);
+    int rval = aws_array_list_comparator_string(&str_a, &str_b);
+    if (!rval) {
+        assert_bytes_match(str_a->bytes, str_b->bytes, str_a->len);
+    }
```

Comment:
Can you make sure that `str_a->bytes` does not match `str_b->bytes` if `!rval == false`?

Reply:
That would require a universal quantifier, rather than an existential qualifier, and would be inefficient.  If we can find a nice way to do it, we should add that in a future PR.